File: common/src/main/java/com/itsmtools/common/dictionary/service/ServiceService.java
package com.itsmtools.common.dictionary.service;


import com.itsmtools.common.dictionary.model.Service;
import com.itsmtools.common.dictionary.repository.ServiceRepository;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.List;
import java.util.Map;


@org.springframework.stereotype.Service
public class ServiceService {

    @Autowired
    ServiceRepository repository;

    public Service item(Integer id) {
        return repository.get(id);
    }

    public void save(Service input) {
        repository.create(input);
    }

    public void update(Service input) {
        repository.update(input);
    }

    public List<Service> list(Map<String, String> filter) {
        return repository.list(filter);
    }
}

File: common/src/main/resources/META-INF/resources/public/application/module/common/dictionary/data/service/service-service.js
function factory($http) {
  return {
    list: function (start, limit) {
      return $http.get('/service', {})
        .then(function (response) {
          return response.data;
        });
    },

    create: function(data){
      return $http.post('/service', data)
        .then(function (response) {
          return response.data;
        });
    },

    update: function(data){
      return $http.put('/service', data)
        .then(function (response) {
          return response.data;
        });
    }
  }
}

export const config = {
  name: "serviceService",
  fn: factory
};